src/openrct2/ride/TrackData.cpp

```diff
@@ -7183,6 +7183,8 @@ namespace OpenRCT2
 
         const TrackElementDescriptor& GetTrackElementDescriptor(const uint32_t type)
         {
+            if (type >= _trackElementDescriptors.size())
+                return _trackElementDescriptors[0];
```

Comment:
Shouldn't this throw instead? This behaviour might hide potential bugs.

Reply:
I agree. Workarounds like this will prevent crashes, but cause side effects instead that are difficult to trace back here.

Reply:
Bad idea. We have dozens of crashes because of nullptrs caused by bad object indices. TED indices are no different. They tend to happen when loading saves using newer track elements in older versions that don’t have them yet.

Reply:
That sounds like crashes because of loading parks that are of a newer version, in which case an exception should be thrown during park load. This is just like the issue that Karst ran into a while ago, where he was running a park that was saved with a debug build that had additional pieces.

I'd much rather see this handled properly, that hiding bugs with null checks and dummy values being used.

Reply:
Define "handled properly", please. It currently crashes or reads garbage data, throwing an error means it will still crash. It’s sticking up a middle finger to anyone unfortunate enough to have a small glitch in the wrong place.

We used to have asserts on ride types too, making for countless crashes bug reports. This is just bringing the TEDs in line with the RTDs.

Reply:
> That sounds like crashes because of loading parks that are of a newer version, in which case **an exception should be thrown during park load**.

When trying to open a park file from a newer build that includes unknown track blocks. This will not cause a crash, this will simply fail to load the save, and the error message can show why.

Reply:
Newly added track pieces can be handled like that, but as I said, these indices get corrupted for other reasons as well. How do you propose loading those saves?

Reply:
Corruption sounds pretty serious and has to be found and fixed then. In any case, loading a corrupt file doesn't make sense - it should error out during load, instead of error during gameplay. If the game manages to open corrupt parks, players may continue building on a broken save without even knowing it.

If it's possible to fix the corruption during load, then that would be even better of course. Loading invalid data and have the game full of dummy exceptions and null checks is no way to move forward.

Reply:
You should always prevent crashing. It should either be defensive here, or validate at load time.

Reply:
@Broxzier Really?! OpenRCT2 has countless null checks for objects/tile elements/etc. because corruption happens *all the damn time*. I actually once created a PR that removed any tile elements that referred to a missing objects and was quickly told it was a very bad idea because it also prevents people from fixing. Blocking a save file with a corrupted track type reference is beyond draconian.

Reply:
I'm sideing with the return the 0 entry side. Now that a few folk are creating new tracks it's a real pain crashing saves between versions. (Ideally we would be bumping the park version with each new track as well though)